client/src/model.js

```diff
@@ -0,0 +1,164 @@
+import { Observable } from 'rxjs/Observable'
+
+// Other imports
+import isPlainObject from 'is-plain-object'
+
+// Project imports
+import { isRecursivelyPrimitive } from './util/is-recursively-primitive'
+
+// Unlike normal queries' .watch(), we don't support rawChanges: true
+// for aggregates
+function checkWatchArgs(args) {
+  if (args.length > 0) {
+    throw new Error(".watch() on aggregates doesn't support arguments!")
+  }
+}
+
+function hasTermInterface(possibleTerm) {
+  return typeof possibleTerm.fetch === 'function' &&
+         typeof possibleTerm.watch === 'function'
+}
+
+function hasObservableInterface(possibleObservable) {
+  return typeof possibleObservable.subscribe === 'function' &&
+         typeof possibleObservable.lift === 'function'
+}
+
+// Simple wrapper for primitives. Just emits the primitive
+class PrimitiveTerm {
+  constructor(value) {
+    this._value = value
+  }
+
+  toString() {
+    return this._value.toString()
+  }
+
+  fetch() {
+    return Observable.of(this._value)
+  }
+
+  watch(...watchArgs) {
+    checkWatchArgs(watchArgs)
+    return Observable.of(this._value)
```

Comment:
@deontologician similar to our discussion previously concerning `Array/AggregateTerm`, a seperate `ObservableTerm` is not strictly necessary. Since `Observable.from(<IObservable>)` is a noop, we could get away with just having a `ValueTerm` to handle everything else (primitives, observables, etc).

Also, looks like the current implementation doesn't support promises, but by using `Observable.from(...)` here we'd get that for free.